# Ollama 一键安装启动
第一次运行选择 >> "重启内核并运行所有单元格"

https://ollama.com/library 查看支持的模型

## 安装 ModelScope 下载器
详见 [Ollama-Linux安装](https://www.modelscope.cn/models/modelscope/ollama-linux)

In [1]:
!pip install modelscope

Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.8/5.8 MB 19.6 MB/s eta 0:00:00a 0:00:01

[notice] A new release of pip is available: 24.0 -> 24.3.1
[notice] To update, run: python -m pip install --upgrade pip


## 从 ModelScope 上下载 Ollama 安装包

In [2]:
%%time
!modelscope download --model=modelscope/ollama-linux --local_dir /root/gpufree-data/ollama-linux --revision v0.5.1

Fetching 3 files:   0%|                                   | 0/3 [00:00<?, ?it/s]





Fetching 3 files:  33%|█████████                  | 1/3 [00:00<00:00,  2.41it/s]


Fetching 3 files: 100%|███████████████████████████| 3/3 [02:38<00:00, 52.94s/it]
CPU times: user 1.51 s, sys: 403 ms, total: 1.91 s
Wall time: 2min 41s


## 安装 ollama

In [3]:
!mkdir -p /root/gpufree-data/ollama-data && ln -s /root/gpufree-data/ollama-data /root/.ollama
!cd /root/gpufree-data/ollama-linux && chmod 777 ./ollama-modelscope-install.sh && ./ollama-modelscope-install.sh

>>> Installing Ollama to /usr/local/bin
>>> Unzipping Ollama-bundle to /usr/local...
>>> Creating ollama user...
>>> Adding ollama user to video group...
>>> Adding current user to ollama group...
>>> Creating ollama systemd service...
>>> systemd is not running
>>> NVIDIA GPU installed.
>>> The Ollama API is now available at 127.0.0.1:11434.
>>> Install complete. Run "ollama" from the command line.


## 启动 Ollama 服务

In [ ]:
!ollama serve

Couldn't find '/root/.ollama/id_ed25519'. Generating new private key.
Your new public key is: 

ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIJr5SzYWpOWxmKWgLGJOiYG3lLWfMABLzF6YYVI/a9Rw

2024/12/12 10:31:19 routes.go:1195: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVICE_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION: HTTPS_PROXY: HTTP_PROXY: NO_PROXY: OLLAMA_DEBUG:false OLLAMA_FLASH_ATTENTION:false OLLAMA_GPU_OVERHEAD:0 OLLAMA_HOST:http://127.0.0.1:11434 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_KV_CACHE_TYPE: OLLAMA_LLM_LIBRARY: OLLAMA_LOAD_TIMEOUT:5m0s OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODELS:/root/.ollama/models OLLAMA_MULTIUSER_CACHE:false OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://localhost https://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http://0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* 

## 运行 Ollama
在终端输入 `ollama run llama3.2`

或者在控制台，选择 **自定义服务** 打开 11434 端口，用 Chatbox 等客户端连接